### ISSUE TITLE
[CW-2602] Webpacker doesn't work properly on Mac M1

### ISSUE THREAD

**Issue by mazenkhalil** [chatwoot/chatwoot#8020]
### Describe the bug

Upon upgrade to Node20, Webpacker doesn't generate the public directory properly. only `manifest.json` file get added to `public/packs` directory.

I believe the issue is related to incompatible dependencies. 

The following warning shows upon run `make setup` command using node20.

```
warning Error running install script for optional dependency: "/Users/abc/Desktop/chatwoot/node_modules/watchpack-chokidar2/node_modules/fsevents, /Users/abc/Desktop/chatwoot/node_modules/webpack-dev-server/node_modules/fsevents: Command failed.
Exit code: 1
Command: node install.js
Arguments: 
Directory: /Users/abc/Desktop/chatwoot/node_modules/webpack-dev-server/node_modules/fsevents
Output:
node:events:492
      throw er; // Unhandled 'error' event
      ^

Error: spawn node-gyp ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess._handle.onexit (node:internal/child_process:292:12)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn node-gyp',
  path: 'node-gyp',
  spawnargs: [ 'rebuild' ]
```

### To Reproduce

1. Clone chatwoot v3.1.1 from github
2. Install node 20.5.1 and run `nvm use`
3. Run `make setup`
4. Observe the warning related to fsevents
5. Run `make db`
6. Run `make run` 
7. Check public/packs directory..

![image](https://github.com/chatwoot/chatwoot/assets/21160900/0f296f0c-4b2e-4a3e-8336-85f226e79dae)

### Environment

Other [please specify in the description]

### Cloud Provider

None

### Platform

Browser

### Operating system

Mac M1

<sub>[CW-2602](https://linear.app/chatwoot/issue/CW-2602/webpacker-doesnt-work-properly-with-node20-on-mac-m1)</sub>

**Comment by scmmishra** [comment]
Can you try reinstalling node-gyp? `npm install -g node-gyp`

**Comment by mazenkhalil** [comment]
The warning disappered, but still the `public/packs` folder has only the manifest.json.

![image](https://github.com/chatwoot/chatwoot/assets/21160900/d6468239-c18c-4b12-94e8-77adaa442b49)

**Comment by scmmishra** [comment]
Is this your dev instance or production? You will need to have the webpack server running in dev mode. You can run Procfile.dev using [Overmind](https://github.com/DarthSim/overmind) or any other process manager

**Comment by mazenkhalil** [comment]
I do run the server with `make run`. The frontend logs appear but only the manifest file get created.

**Comment by mazenkhalil** [comment]
This really weird... upon I run `export NODE_OPTIONS=--openssl-legacy-provider` explicitly on the terminal, the `public/packs` directory get populated, but the browser couldn't serve the request (I tried multiple browsers same problem)

![image](https://github.com/chatwoot/chatwoot/assets/21160900/ec58a3a0-c1b7-463f-be39-cebc5ac2b986)

<img width="1507" alt="image" src="https://github.com/chatwoot/chatwoot/assets/21160900/3c6c7930-0bd8-4339-b1cc-7abb6c06de23">

**Comment by mazenkhalil** [comment]
Hello @scmmishra, any update on this. It is almost impossible to do development on Mac. The behavior is inconsistent. any workaround would be great.

To summarize, the webpacker doesn't behave correctly on Mac. The only way to run a development environment is by disabling the web packer which makes frontend development a disaster.

**Comment by sojan-official** [comment]
@mazenkhalil are you running the default Chatwoot repo without any customisations? Everyone in the Chatwoot team uses Mac for development and there haven't been any issues.

**Comment by mazenkhalil** [comment]
@sojan-official  Default repo, clean installation on multiple MAC devices (Apple Silicon) same issue...

**Comment by mazenkhalil** [comment]
Yes, this issue also occurs in the latest version. 

While debugging, I discovered that requests for loading assets, such as `http://localhost:3000/packs/js/17-a21f726f767654ef7d3a.chunk.js`, are sometimes served and sometimes result in a `402` error code.

It is important to note that accessing the same asset using the dev-server port `3035` always succeeds.

After some trial and error, I managed to resolve the problem by changing the dev-server host from `0.0.0.0` to `127.0.0.1`/`localhost`.

![image](https://github.com/chatwoot/chatwoot/assets/21160900/d611e0da-4290-42b6-9ad3-7c955043b5a2)

I am unsure why the issue arises, but this adjustment effectively resolved it, even though the port `3035` is not being used by other services.

I am documenting this solution in case someone else faces a similar issue.
